internal/driver/onvifdiscovery.go

```diff
@@ -0,0 +1,235 @@
+	for _, dev := range devices {
+		res = append(res, netscan.ProbeResult{
+			Host: host,
+			Port: port,
+			Data: dev,
+		})
+	}
+	return res
+}
+
+// executeRawProbe essentially performs a UDP unicast ws-discovery probe by sending the
+// probe message directly over the connection and listening for any responses. Those
+// responses are then converted into a slice of onvif.Device.
+func executeRawProbe(conn net.Conn, params netscan.Params) ([]onvif.Device, error) {
+	probeSOAP := wsdiscovery.BuildProbeMessage(uuid.Must(uuid.NewV4()).String(), nil, nil,
+		map[string]string{"dn": "http://www.onvif.org/ver10/network/wsdl"})
+
+	addr := conn.RemoteAddr().String()
+	if err := conn.SetDeadline(time.Now().Add(params.Timeout)); err != nil {
+		return nil, errors.Wrapf(err, "%s: failed to set read/write deadline", addr)
+	}
+
+	if _, err := conn.Write([]byte(probeSOAP.String())); err != nil {
+		return nil, errors.Wrap(err, "failed to write probe message")
+	}
+
+	var responses []string
+	buf := make([]byte, bufSize)
+	// keep reading from the PacketConn until the read deadline expires or an error occurs
+	for {
+		n, _, err := (conn.(net.PacketConn)).ReadFrom(buf)
+		if err != nil {
+			// ErrDeadlineExceeded is expected once the read timeout is expired
+			if !errors.Is(err, os.ErrDeadlineExceeded) {
+				params.Logger.Debugf("Unexpected error occurred while reading ws-discovery responses: %s", err.Error())
+			}
+			break
+		}
+		responses = append(responses, string(buf[0:n]))
+	}
+
+	if len(responses) == 0 {
+		params.Logger.Tracef("%s: No Response", addr)
+		return nil, nil
+	}
+	for i, resp := range responses {
+		params.Logger.Debugf("%s: Response %d of %d: %s", addr, i+1, len(responses), resp)
+	}
+
+	devices, err := wsdiscovery.DevicesFromProbeResponses(responses)
+	if err != nil {
+		return nil, err
+	}
+	if len(devices) == 0 {
+		params.Logger.Debugf("%s: no devices matched from probe response", addr)
+		return nil, nil
+	}
+
+	return devices, nil
+}
+
+// makeDeviceMap creates a lookup table of existing devices by EndpointRefAddress
+// todo: will be used in the future for device re-discovery purposes
+//func (d *Driver) makeDeviceMap() map[string]contract.Device {
+//	devices := d.svc.Devices()
+//	deviceMap := make(map[string]contract.Device, len(devices))
+//
+//	for _, dev := range devices {
+//		if dev.Name == d.serviceName {
+//			// skip control plane device
+//			continue
+//		}
+//
+//		onvifInfo := dev.Protocols[OnvifProtocol]
+//		if onvifInfo == nil {
+//			d.lc.Warnf("Found registered device %s without %s protocol information.", dev.Name, OnvifProtocol)
+//			continue
+//		}
+//
+//		endpointRef := onvifInfo["EndpointRefAddress"]
+//		if endpointRef == "" {
+//			d.lc.Warnf("Registered device %s is missing required %s protocol information: EndpointRefAddress.",
+//				dev.Name, OnvifProtocol)
+//			continue
+//		}
+//
+//		deviceMap[endpointRef] = dev
+//	}
+//
+//	return deviceMap
+//}
```

Comment:
Remove commented out code

Reply:
only commented out to keep the linter happy. the code is in use in the rediscovery PR. Now that it is being used there I can remove it here and not lose it.